### PR TITLE
Page translation links

### PR DESCRIPTION
**Pull request by chosak**
This PR adds the ability to link translated Wagtail pages and have those links automatically rendered atop those pages. It also adds a basic Wagtail report for reviewing which pages have translated versions, as well as a simple script for importing page translation links in bulk. This implements the design described on internal D&CP#122.

(Note that this approach is not the same as Wagtail's [built-in approach](https://docs.wagtail.org/en/stable/advanced_topics/i18n.html) to such things.) 

### Page translation links

Pages now have a new `english_page` field, which is intended for use on non-English pages, to link back to the English page of which they are a translation. For example, given a page A in English and a page B in Spanish that is a translation of page A, B's `english_page` would point to A. English pages should not use this field, although nothing terrible will happen if it is set.

Pages now also have a `translations` property which returns a PageQuerySet of all pages that are translations of that page. This can be used on all pages regardless of language.

### Page editor interface

This new field can now be set in the Wagtail admin page editor interface as part of a new consolidated "Translation" section that can be found on pages' "Configuration" tab:

image

### Bulk import of translation links

To make it easier to test and implement this feature, this PR includes a new Django management command, `import_translation_links`, that allows for loading of page links from a CSV. See `import_translation_links.py` in this PR for specific implementation details and expected CSV format.

### "Translated Pages" report

This PR adds a new, very basic, [Wagtail report](http://localhost:8000/admin/reports/translated-pages/) for reviewing page translations. It's very basic but allows for filtering by translation language:

image

### Automatic rendering of translation links

For now I've only implemented a first cut at rendering translation links as part of the Item Introduction module, which is used on most blog pages. A good representative production URL to test with locally is [this one](http://localhost:8000/about-us/blog/hidden-cost-junk-fees-ar/), which looks like this (links are duplicated with the existing hardcoded rich text ones):

image

Obviously the links aren't in the right place, and there are some other questions: should the language links be translated into the page's language, or not? One thing I've implemented here is the use of `hreflang` in the links, which partially addresses internal XP#13.

## How to test this PR

All code changes should be covered by unit tests with (I hope) total coverage.

## Notes and todos

- Page editor interface: it would be nice if the interface were more dynamic, for example hiding or disabling the "English page" field if the page's language is English. I'm not sure how possible this is; it would probably be better to wait until we are on Wagtail 4 to do any frontend customization like this.
- "Translated Pages" report: there's a lot more to do here; for example the export doesn't currently include any translation information. For the to be truly useful it might need to include page URLs, though, which we've had trouble with in the past. The current layout (combining all translations in one column) is just an initial approach; open to any ideas @csebianlander.
- Automatic rendering of translation links: before we can move forward with this we'd need to figure out how to practically turn this on at the same time as we remove the current hardcoded links. One option would be to hide this behind a feature flag turned off by default while we iterate on design and implementation questions.
- Documentation needs to be added for this, both internal and [external](https://cfpb.github.io/consumerfinance.gov/translation/).

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### CODE REVIEW

**Comment by csebianlander** [comment]
I love this. Back-end implementation (in terms of content editor experience) is pretty straightforward and should be easy to document for expanded use.

Thoughts in general and on some of your notes:

> Page editor interface: it would be nice if the interface were more dynamic, for example hiding or disabling the "English page" field if the page's language is English. I'm not sure how possible this is; it would probably be better to wait until we are on Wagtail 4 to do any frontend customization like this.

yeah this is very low-priority to me, especially since if the page is in English then you're probably not even needing to look at the Translation section of the tab at all.

> "Translated Pages" report: there's a lot more to do here; for example the export doesn't currently include any translation information. For the to be truly useful it might need to include page URLs, though, which we've had trouble with in the past. The current layout (combining all translations in one column) is just an initial approach; open to any ideas @csebianlander.

I need to give this some thought, but it's OK to just have it as-is for this initial step and then we can expand on it in a separate task.

> Automatic rendering of translation links: before we can move forward with this we'd need to figure out how to practically turn this on at the same time as we remove the current hardcoded links. One option would be to hide this behind a feature flag turned off by default while we iterate on design and implementation questions.

So, considering we're still in-progress on figuring out styling approaches, maybe for now we can just hide this on the front-end (maybe commenting out or removing everything from `/molecules/translations.html`, but keeping the infrastructure to re-add it still in place). Then we still get the ability to set up the translations on the back-end, which should make it faster and easier to go through and remove hardcoded links ahead of additional work to implement a short-term visual solution that works.

**Comment by chosak** [comment]
Thanks for the feedback @csebianlander.

>> Page editor interface: it would be nice if the interface were more dynamic, for example hiding or disabling the "English page" field if the page's language is English. I'm not sure how possible this is; it would probably be better to wait until we are on Wagtail 4 to do any frontend customization like this.
>
> yeah this is very low-priority to me, especially since if the page is in English then you're probably not even needing to look at the Translation section of the tab at all.

It had also occurred to me that it might be useful to use this editor section to be able to see the translations as well, though, for example to be able to directly click to edit one. The need for this becomes somewhat less on Wagtail 4 with the live sidebar preview, though.

Any thoughts on whether we should move the language selection bits to a separate Translation tab in the editor, vs. including it on the Configuration tab?

**Comment by csebianlander** [comment]
yeah I think we've discussed that option before - I think of it as a nice-to-have but not necessary for this iteration.

I'm OK with keeping the Translation item as it currently stands within Configuration, but if we _did_ implement a way to view the translations right on the page, it'd be nice to separate that out for better visibility.

File: cfgov/v1/jinja2/v1/includes/organisms/item-introduction.html [hunk 1]
```diff
@@ -40,6 +40,8 @@
     {% endif %}
     <h1>{{ value.heading | safe }}</h1>
 
+    {% include 'v1/includes/molecules/translations.html' %}
```
Comment (willbarton) [inline cfgov/v1/jinja2/v1/includes/organisms/item-introduction.html:43]: ***If*** we wanted to limit this PR to just enabling the admin-side of this functionality for now, we could wrap this in a feature flag.

**Review by willbarton — APPROVED** [review]
I think this looks great!

File: cfgov/v1/management/commands/import_translation_links.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Import page translation links from a CSV file with format:
+#
+#    translated_page_relative_url,translated_page_english_url
```
Comment (csebianlander) [inline cfgov/v1/management/commands/import_translation_links.py:3]: should the URL provided in translated_page_english_url be relative or absolute?

**Review by csebianlander — APPROVED** [review]
Once we implement some of the short-term design things we've been discussing into `translations.html` I think this is ready to go - the content manager experience all works as expected.

File: cfgov/v1/management/commands/import_translation_links.py (hunk 2 above)
Comment (chosak) [reply cfgov/v1/management/commands/import_translation_links.py:3]: Both should be relative! I'll update the script.

**Comment by chosak** [comment]
Thanks @willbarton @csebianlander for the reviews - I'm going to push some additional changes to simplify this and get this mergeable behind a feature flag as we discussed, and then I'll move this out of draft status and re-request review.

**Comment by chosak** [comment]
@csebianlander @willbarton this is ready for review again.

I've put any visible frontend changes behind a `TRANSLATION_LINKS` feature flag which can be enabled locally by going to http://localhost:8000/admin/flags/TRANSLATION_LINKS/.

Once enabled, you can preview the current link appearance at http://localhost:8000/admin/template_debug/v1/translation_links/.

I've added some pages to our test database to make it easy to test the current functionality without needing to mess with production data; you should be able to load the test database using `./refresh-data.sh test.sql.gz` and then navigating to http://localhost:8000/about-us/blog/ to see and edit the linked pages.

There is more work to do before this feature can go live, but this PR is intended to be a foundation upon which we can build further.

**Review by willbarton — APPROVED** [review]

**Review by csebianlander — APPROVED** [review]

**Comment by csebianlander** [comment]
Some of the ordering of links is weird, like on the English test page:
Screen Shot 2023-01-20 at 12 31 00 PM

I assume this is because of the incomplete native language/alphabetizing support?

Works beautifully otherwise!

**Comment by chosak** [comment]
> Some of the ordering of links is weird

@csebianlander yes, this needs to be addressed along with the translations.

Thank you both for the reviews!

**Pull request merged**